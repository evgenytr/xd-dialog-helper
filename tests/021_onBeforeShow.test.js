const dialogHelper = require('../src/main');

let resolver, rejector;

HTMLElement.prototype.showModal = () => {
    return new Promise((resolve, reject) => {
        resolver = resolve;
        rejector = reject;
    });
};
HTMLElement.prototype.close = (value) => {
    if (value !== 'reasonCanceled')
        resolver(value);
    else
        rejector(value);
};

describe('options.onBeforeShow', () => {
    it('should get called', () => {
        const beforeFn = jest.fn(() => true);
        dialogHelper.showDialog('dialog', 'My dialog', [], {
            onBeforeShow: beforeFn
        });

        expect(beforeFn).toBeCalled();
    });

    it('should get passed the correct parameters', () => {
        const beforeFn = jest.fn(() => true);

        const props = {
            id: 'cb',
            type: dialogHelper.types.CHECKBOX,
            value: false,
            required: true,
            label: 'Checkbox'
        };

        dialogHelper.showDialog('dialog', 'My dialog', [
            props
        ], {
            onBeforeShow: beforeFn
        });

        expect(beforeFn).toBeCalled();

        const args = beforeFn.mock.calls[0];

        expect(args[1][0].props).toBe(props);
        expect(document.body.children).toContain(args[0]);
        expect(args[1][0].input.id).toBe('dialog-cb');

        expect(beforeFn).toMatchSnapshot();
    });
});
