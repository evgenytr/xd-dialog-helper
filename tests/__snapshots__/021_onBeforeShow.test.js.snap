// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`options.onBeforeShow should get passed the correct parameters 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      <dialog
        id="dialog-dialog"
      >
        

        <style>
          
    dialog#dialog {
            width: 360px;
    }
        
    undefined

        </style>
        

        <form
          method="dialog"
        >
          

          <header>
            <h1>
              My dialog
            </h1>
            <hr />
          </header>
          

          <main>
            <label
              id="dialog-cb-wrapper"
              style="flex-direction: row; align-items: center;"
            >
              <input
                id="dialog-cb"
                placeholder="Checkbox"
                type="checkbox"
                value="false"
              />
              <span
                id="dialog-cb-label"
              >
                Checkbox
              </span>
            </label>
          </main>
          

          <footer>
            

            <button
              id="dialog-dialogHelperBtnCancel"
              uxp-variant="primary"
            >
              Cancel
            </button>
            

            <button
              id="dialog-dialogHelperBtnOk"
              type="submit"
              uxp-variant="cta"
            >
              Ok
            </button>
            

          </footer>
          

        </form>
        
    
      </dialog>,
      Array [
        Object {
          "input": <input
            id="dialog-cb"
            placeholder="Checkbox"
            type="checkbox"
            value="false"
          />,
          "props": Object {
            "id": "cb",
            "label": "Checkbox",
            "required": true,
            "type": Object {
              "render": [Function],
              "type": "Input",
              "valid": [Function],
              "value": [Function],
            },
            "value": false,
          },
          "type": Object {
            "render": [Function],
            "type": "Input",
            "valid": [Function],
            "value": [Function],
          },
          "wrapper": <label
            id="dialog-cb-wrapper"
            style="flex-direction: row; align-items: center;"
          >
            <input
              id="dialog-cb"
              placeholder="Checkbox"
              type="checkbox"
              value="false"
            />
            <span
              id="dialog-cb-label"
            >
              Checkbox
            </span>
          </label>,
        },
      ],
      Object {
        "cancel": [Function],
        "change": [Function],
        "close": [Function],
        "registerContentElement": [Function],
        "values": [Function],
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": true,
    },
  ],
}
`;
